File: app/src/main/java/com/kgb/k_app/data/DataSource.java
package com.kgb.k_app.data;

import java.util.List;

/**
 * Created by Jan on 27.11.2016.
 */

public interface DataSource<Model extends Data> {
    void confirmedChanges(Model model);

    List<Model> retrieveData();

    List<Model> retrieveData(String where);

    int count();

    Model get(int position);

    long getItemId(int position);
}

File: app/src/main/java/com/kgb/kapp/data/DataSource.kt
package com.kgb.kapp.data

interface DataSource<Model : Data> {
    fun confirmedChanges(model: Model)

    fun retrieveData(): List<Model>

    fun retrieveData(where: String): List<Model>

    fun count(): Int

    operator fun get(position: Int): Model

    fun getItemId(position: Int): Long
}